### PR TITLE
docs: Go Snippets

### PR DESCRIPTION
**Pull request by Anush008**
### NOTE

The `fern/apis` files are auto generated.

### CODE REVIEW

File: snippets/go/clear_payload.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	"github.com/qdrant/go-client/qdrant"
+)
+
+func clearPayload() {
+	client, _ := qdrant.NewClient(&qdrant.Config{
+		Host: "localhost",
+		Port: 6334,
+	})
+
+	client.ClearPayload(context.Background(), &qdrant.ClearPayloadPoints{
```
Comment (Robert-Stam) [inline snippets/go/clear_payload.go:15]: same here

File: snippets/go/collection_exists.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	"github.com/qdrant/go-client/qdrant"
+)
+
+func collectionExists() {
+	client, _ := qdrant.NewClient(&qdrant.Config{
+		Host: "localhost",
+		Port: 6334,
+	})
+
+	client.CollectionExists(context.Background(), "{collection_name}")
```
Comment (Robert-Stam) [inline snippets/go/collection_exists.go:15]: and here

File: snippets/go/clear_payload.go (hunk 1 above)
Comment (Anush008) [reply snippets/go/clear_payload.go:15]: I thought of checking the error.
But ignored it to keep the snippet a little concise.

**Review by Robert-Stam — COMMENTED** [review]
See remark, this applies to the majority of code.

IMO It's a bad idea to ignore an error in a sample, customers will copy-paste some of the snippets and will not add it (unless they have configured a linter which detect it).

**Comment by Anush008** [comment]
I thought of checking the error.
But ignored it to keep the snippets a little concise.

@Robert-Stam, how about we just panic with the error?

File: snippets/go/batch_update.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	"github.com/qdrant/go-client/qdrant"
+)
+
+func batchUpdate() {
+	client, _ := qdrant.NewClient(&qdrant.Config{
+		Host: "localhost",
+		Port: 6334,
+	})
+
+	client.UpdateBatch(context.Background(), &qdrant.UpdateBatchPoints{
```
[the remark this answers is not in the document]
Comment (Robert-Stam) [reply snippets/go/batch_update.go:18]: I think we should not ignore the error from UpdateBatch as well
image

[the remark this answers is not in the document]
Comment (Robert-Stam) [reply snippets/go/batch_update.go:18]: (panic is fine, like)
```
if _, err := client.UpdateBatch(context.Background(), &qdrant.UpdateBatchPoints{
...
}); err != nil {
  panic(err)
  }

**Review by Robert-Stam — COMMENTED** [review]
See comment

File: snippets/go/batch_update.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Anush008) [reply snippets/go/batch_update.go:18]: > if this is returning an error, maybe check it?
All of them do actually. An operation result and error.

[the remark this answers is not in the document]
Comment (Anush008) [reply snippets/go/batch_update.go:18]: We'll have use the result and error too.
So didn't use both.

[the remark this answers is not in the document]
Comment (Robert-Stam) [reply snippets/go/batch_update.go:18]: Same applies here: IMO It's a bad idea to ignore an error in a sample, customers will copy-paste some of the snippets and will not add it (unless they have configured a linter which detect it).

[the remark this answers is not in the document]
Comment (Anush008) [reply snippets/go/batch_update.go:18]: We also use the value?

[the remark this answers is not in the document]
Comment (Anush008) [reply snippets/go/batch_update.go:18]: Or as you suggested, ignore it?

[the remark this answers is not in the document]
Comment (Robert-Stam) [reply snippets/go/batch_update.go:18]: If there is something relevant in the result, we should add it to the sample, if it's an empty struct, there is no needed to do imo.

[the remark this answers is not in the document]
Comment (Anush008) [reply snippets/go/batch_update.go:18]: The problem is we have to use the value if we declare it. Even if doesn't contain something useful.
Or else we get
`declared and not used: snapshot`

[the remark this answers is not in the document]
Comment (Anush008) [reply snippets/go/batch_update.go:18]: Some of the responses only contain response times.

[the remark this answers is not in the document]
Comment (Robert-Stam) [reply snippets/go/batch_update.go:18]: It should work with a _

**Review by Robert-Stam — APPROVED** [review]
LGTM

**Pull request merged**